### PR TITLE
Support Vault KV version 2 and unique secret names

### PR DESCRIPTION
**Pull request by bonifaido**
Fixes: #553

### CODE REVIEW

File: secret/store.go [hunk 1]
```diff
@@ -79,9 +86,8 @@ func newVaultSecretStore() *secretStore {
 	return &secretStore{client: client, logical: logical}
 }
 
-// GenerateSecretID uuid for new secrets
-func GenerateSecretID() string {
-	return uuid.NewV4().String()
+func generateSecretID(request *CreateSecretRequest) string {
+	return base64.StdEncoding.EncodeToString([]byte(request.Name))
```
Comment (baluchicken) [inline secret/store.go]: Maybe a good Hash function (Sha-1, Sha-256) is a better choice:)

Comment (stoader) [reply secret/store.go]: sha-256 should be good enough

Comment (bonifaido) [reply secret/store.go]: So true!

**Comment by bonifaido** [comment]
I will add also Postman and Swagger updates.

**Comment by bonifaido** [comment]
Ready for review.

**Pull request merged**